File: streaming/src/main/java/nl/ns/jfr/JFRInfluxStreamer.java
package nl.ns.jfr;

import com.influxdb.client.InfluxDBClient;
import com.influxdb.client.InfluxDBClientFactory;
import com.influxdb.client.domain.WritePrecision;
import com.influxdb.client.write.Point;
import io.quarkus.runtime.Startup;
import java.time.temporal.ChronoUnit;
import javax.enterprise.context.ApplicationScoped;
import jdk.jfr.RecordingState;
import jdk.jfr.consumer.RecordingStream;

@Startup
@ApplicationScoped
public class JFRInfluxStreamer {

    private final InfluxDBClient influxDBClient = InfluxDBClientFactory.create(
            "http://localhost:8086", "token".toCharArray(), "NS", "JFR"
    );

    public JFRInfluxStreamer() {
        RecordingStream recordingStream = new RecordingStream();
        recordingStream.enable(RestCallEvent.class);

        recordingStream.onEvent(recordedEvent -> {
            influxDBClient.getWriteApi()
                    .writePoint(
                            new Point("RestCallEvent")
                                    .time(recordedEvent.getStartTime().toEpochMilli(), WritePrecision.MS)
                                    .addField("duration", recordedEvent.getDuration().get(ChronoUnit.MILLIS))
                                    .addField("endpoint",recordedEvent.getValue("endpoint").toString())
                                    .addField("ip", recordedEvent.getValue("ip").toString()));
        });

        recordingStream.startAsync();
    }
}

File: streaming/src/main/kotlin/nl/ns/jfr/JFRInfluxStreamer.kt
package nl.ns.jfr

import com.influxdb.client.domain.WritePrecision
import com.influxdb.client.kotlin.InfluxDBClientKotlinFactory
import com.influxdb.client.write.Point
import jdk.jfr.consumer.RecordingStream
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch

class JFRInfluxStreamer {
    private val influxDBFactory = InfluxDBClientKotlinFactory.create(
        "http://localhost:8086", "token".toCharArray(), "NS", "JFR"
    )
    init {
        val recordingStream = RecordingStream()
        recordingStream.enable(RestCallEvent::class.java)

        recordingStream.onEvent {
            val writeApi = influxDBFactory.getWriteKotlinApi()
            GlobalScope.launch {
                writeApi.writePoint(
                    Point.measurement("rest")
                        .time(it.startTime.toEpochMilli(), WritePrecision.MS)
                        .addField("duration", it.duration.nano)
                        .addTag("endpoint", it.getString("endpoint"))

                )
            }
        }

        recordingStream.startAsync()
    }


}